tests/Microsoft.Identity.Test.Unit/ManagedIdentityTests/ImdsV2Tests.cs

```diff
@@ -130,5 +136,39 @@ public async Task GetCsrMetadataAsyncFails404WhichIsNonRetriableAndRetryPolicyIs
                 Assert.AreEqual(ManagedIdentitySource.DefaultToImds, miSource);
             }
         }
+
+        [TestMethod]
+        public void TestCsrGeneration()
+        {
+            var cuid = new CuidInfo
+            {
+                VmId = TestConstants.VmId,
+                VmssId = TestConstants.VmssId
+            };
+
+            // Generate CSR
+            var csrPem = Csr.Generate(TestConstants.ClientId, TestConstants.TenantId, cuid);
+
+            // Validate the CSR contents using the helper
+            CsrValidator.ValidateCsrContent(csrPem, TestConstants.ClientId, TestConstants.TenantId, cuid);
+        }
+
+        [TestMethod]
+        public void TestCsrGeneration_MalformedPem_FormatException()
```

Comment:
Can we also add a positive tests for ParseCsrFromPem on well-formed input?

Reply:
Can you please point reviewers to that test.

Reply:
I believe that the existing TestCsrGeneration tests should cover this